core/src/io/anuke/mindustry/entities/type/Player.java

```diff
@@ -490,6 +494,10 @@ public void drawBuildRequests(){
     @Override
     public void update(){
 
+        if((unitGroups[waveTeam.ordinal()].size() == 0) && state.rules.limitedRespawns){
```

Comment:
(refers to comment on line 522, 523)
```suggestion
if((unitGroups[waveTeam.ordinal()].size() == 0) && state.rules.respawns != 0){
```